src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -4035,5 +4035,152 @@ .maxstack  2
+                );
+        }
+
+        [Fact]
+        [WorkItem(50126, "https://github.com/dotnet/roslyn/issues/50126")]
+        public void NestedInitializer_UsingGenericType()
+        {
+            var source = @"
+using System;
+
+Person person = new Person(""j"", ""p"");
+var c = new Container<Person>(person)
+{
+    PropertyT = { FirstName = ""c"" }
+};
+
+public record Person(String FirstName, String LastName);
+public record Container<T>(T PropertyT) where T : Person;
+";
+            var comp = CreateCompilation(new[] { IsExternalInitTypeDefinition, source }, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics(
+                // (7,19): error CS8852: Init-only property or indexer 'Person.FirstName' can only be assigned in an object initializer, or on 'this' or 'base' in an instance constructor or an 'init' accessor.
+                //     PropertyT = { FirstName = "c" }
+                Diagnostic(ErrorCode.ERR_AssignmentInitOnly, "FirstName").WithArguments("Person.FirstName").WithLocation(7, 19)
+                );
+        }
+
+        [Fact]
+        [WorkItem(50126, "https://github.com/dotnet/roslyn/issues/50126")]
+        public void NestedInitializer_UsingNew()
+        {
+            var source = @"
+using System;
+
+Person person = new Person(""j"", ""p"");
+Container c = new Container(person)
+{
+    Person = new Person(""j"", ""p"") { FirstName = ""c"" }
+};
+
+Console.Write(c.Person.FirstName);
+
+public record Person(String FirstName, String LastName);
+public record Container(Person Person);
+";
+            var comp = CreateCompilation(new[] { IsExternalInitTypeDefinition, source }, options: TestOptions.DebugExe);
+            comp.VerifyEmitDiagnostics();
+            // PEVerify: Cannot change initonly field outside its .ctor.
+            CompileAndVerify(comp, expectedOutput: "c", verify: ExecutionConditionUtil.IsCoreClr ? Verification.Passes : Verification.Fails);
+        }
+
+        [Fact]
+        [WorkItem(50126, "https://github.com/dotnet/roslyn/issues/50126")]
+        public void NestedInitializer_UsingNewNoPia()
+        {
+            string pia = @"
+using System;
+using System.Runtime.InteropServices;
+
+[assembly: ImportedFromTypeLib(""GeneralPIA.dll"")]
+[assembly: Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58257"")]
+
+[ComImport()]
+[Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58277"")]
+[CoClass(typeof(ClassITest28))]
+public interface ITest28
+{
+    int Property { get; init; }
+}
+
+[Guid(""f9c2d51d-4f44-45f0-9eda-c9d599b58278"")]
+public abstract class ClassITest28 //: ITest28
+{
+    public ClassITest28(int x) { }
+}
+";
+
+            var piaCompilation = CreateCompilationWithMscorlib45(new[] { IsExternalInitTypeDefinition, pia }, options: TestOptions.DebugDll);
+
+            CompileAndVerify(piaCompilation);
+
+            string source = @"
+class UsePia
+{
+    public static void Main()
+    {
+        var x1 = new ITest28();
```

Comment:
The property doesn't appear to actually be being used.

Reply:
Also, do we have a NoPia test doing a legal init assignment?

---
In reply to: [554113056](https://github.com/dotnet/roslyn/pull/50200#discussion_r554113056) [](ancestors = 554113056)

Reply:
Yes, it doesn't look like this unit-test actually tests something interesting. 

---
In reply to: [554113056](https://github.com/dotnet/roslyn/pull/50200#discussion_r554113056) [](ancestors = 554113056)